mixins/compose-classes.js

```diff
@@ -49,8 +49,16 @@ export function composeClasses(baseName, statuses, options) {
       fns.push(seedClasses(baseName, statuses));
     }
   }
+  
+  return (n, sx) => filterUnique(fns.map((fn) => fn(n, sx))).join(' ');
+}
 
-  return (n, sx) => fns.map((fn) => fn(n, sx)).join(' ');
+function filterUnique(arr = []) {
+  var u = {};
+  return arr.filter(function (str) {
+    if (u[str]) return false;
+    return u[str] = 1;
+  });
```

Comment:
it would probably be faster written:

```js
  return applyFns.bind(null, fns);
}

function applyFns(fns, n, sx) {
  var classes = {};
  fns.forEach((fn) => classes[fn(n, sx)] = true);
  return Object.keys(classes).join(' ');
}
```